### DISCUSSION TITLE
I have a vape that’s a VP vape and i’d like to ask on how i can run some games on it

### DISCUSSION THREAD

**Discussion by severedflowers** [ginbot86/ColorLCDVape-RE#4] [General]
since it has three, tetris, f22 top down arcade game and pac-man, maybe we could get a tomogochi or doom running on it? i wanna know how i can connect it to my pc, I have asked around Reddit and they had pointed to your blog about the windows vape

**Comment by ginbot86** [comment]
I noticed some threads on Reddit discussing such:

- https://www.reddit.com/r/hacking/comments/1edi26q/how_do_i_hack_this_to_put_doom_on_it/
- https://www.reddit.com/r/hacking/comments/1edze1h/update_the_correct_photo_was_replaced_by_a_glitch/

Given that the main chip in question is a blob-top/COB (chip on board) device, that will significantly hinder attempts to figure out what it is. However, the [Bootleg Games wiki](https://bootleggames.fandom.com/wiki/V-Play_Vapentainment_System) on Fandom says it's a Famiclone (Nintendo NES clone). Assuming that this is true, this opens and closes a lot of avenues for homebrew software development.

Doom (at least in its pure form) is almost certainly out of the question. However, it might be possible to inject custom NES ROMs into the SPI Flash chip, but the structure of the chip's data needs to be ascertained first. Connecting it to a PC, if it doesn't already show up, may mean that the only way to access the Flash chip's contents is by connecting it to a dedicated device like a universal programmer tool.

**Comment by Rea-XEVION** [comment]
I've got 3 of these v-play vapes and 3 Raz DC25000's, still need to get whatever universal debugger I can find on eBay between $10 and $60.
One thing I quickly noticed is that with the v-play's juice-sponge and coils removed there's enough space for 2x 18650's side by side, so I might try dumping as much as I can from it after learning how with the gutted Raz.

Edit, just noticed that all 3 v-play units run the gameplay at mildly different speeds. For example, 1 and 2 are barely out of sync where you don't notice until a good 10 or 12 seconds later, same for 2 and 3, but comparing 1 and 3 it's easy to notice the desync even just 3 or 4 seconds in.

**Reply by ginbot86** [reply]
The timing differences are probably because the system is running off an internal RC oscillator for its timebase, scaled up or down as required. These oscillators are voltage- and temperature-sensitive, and can have significant variations between units. As a cost-sensitive device I doubt the manufacturer would want to shell out for a ceramic resonator or quartz crystal/oscillator.

**Reply by Rea-XEVION** [reply]
After having 2 v-plays taken apart and still gaming for a few days, I've noticed that changing screens in the vape mode menu just blanks the display when changing screens, same for entering and selecting a game to bring up the sound option. But when actually launching any game after saying yes/no so audio, it completely powers off the screen and backlight for a fraction of a second.
I won't be able to properly dig deep until I buy something more useful than a PICkit 3, but I have to wonder if the main processor is toggling into an A/V pass through mode for the famiclone... Since I have 3 and one's battery broke off, I might just start with "destructive investigation" by cutting one trace at a time, testing if it boots and plays, then repairing and moving on to the next.

If I can find any famiclone specific av traces, or even button traces, I'll try and bodge in an external controller or external av source (like a GameCube)

**Reply by Rea-XEVION** [reply]
I didn't notice this earlier with the "pcb trimming" v-play due to the fact I hadn't used it enough to change from the 100% juice image to the 90% image, but with the unit I refilled a couple times I just noticed it most likely uses the sram for keeping track of puff count.
When the battery positive wire came disconnected I decided not to solder it back on, but instead use my metal craft blade to temporarily jumper the connection for each use of the vape portion.  In doing so, I noticed that what previously read as 80% battery and 0% juice was reset to 100 / 100, though just a moment later it caught up and became 80 battery /100 juice.

**Reply by ginbot86** [reply]
Thanks for the update. Some other vapes I've taken apart that use segmented LED screens seem to hold the vape counter in internal Flash (no external chips onboard), but I haven't dug too deep into where/how they're storing that data.

**Reply by LeojGH** [reply]
I don't know if it helps, but here are some pictures of the V-Play PCB and the model of the external SpiFlash chip (Winbond-25Q32JVSIQ). And what my programmer (CH341A) read from the chip. I am not familiar with these topics but reading your post about Reverse Engineering a Vape to customize the backgrounds of the screen, I was very attentive and then I was curious once I saw that these VPlay came out.
![IMG_20241002_013059](https://github.com/user-attachments/assets/551d602a-9b53-465c-80b8-210c8333b16d)
![IMG_20241002_024747](https://github.com/user-attachments/assets/0a7d5dee-10ab-4159-9ef6-2b6b9ec3f565)
![IMG_20241002_024824](https://github.com/user-attachments/assets/ca717bdb-f6b8-48f2-8f06-4610955f7a2f)
![IMG_20241002_024911](https://github.com/user-attachments/assets/d150db18-76bf-4c4a-b56e-92ebbb5c5dde)
![IMG_20241002_025022](https://github.com/user-attachments/assets/382a4673-5592-481d-857e-21cf64129f8d)
![IMG_20241002_025045](https://github.com/user-attachments/assets/b6e48aac-4c11-4935-a865-72f540a12b33)
[V-PLAY.zip](https://github.com/user-attachments/files/17221685/V-PLAY.zip)

**Reply by ginbot86** [reply]
Thanks for the pictures and the SPI dump. A few observations: The USB-C port on its own board has only power and ground connections, so that would explain why connecting to a PC didn't do anything; I do see a bunch of programming/test pads on J1 (C/D/R = clock, data, reset) for some microcontroller on the board, probably U7; and the P0~P7 labeled pads seems like some kind of parallel data bus (maybe the LCD?).

I looked at the SPI Flash dump and couldn't see anything that immediately stood out to me. It doesn't look like there are any recognizable text strings, ROM file headers, and it might be encrypted or at least scrambled in some way. There's no empty space visible but there seem to be some patterns to the data inside.

**Reply by Rea-XEVION** [reply]
kind of fun that a 6p4c phone /fax/ modem jack can be mounted to the debugging pins.  If I had an smd 6p6c connector then I wouldn't need to use the ground plane for the 5th wire.
Still need to setup a spare laptop with the different ide's for hardware I got months ago and never used
![IMG_20250427_193947.jpg](https://github.com/user-attachments/assets/b152597d-6ebf-4b11-9f35-119bebe6a718)

Edit about 19 hours or 20 hours later. After soldering jumper ends onto a phone cable and plugging into the vPlay, I thought it'd be a good idea to get used to which wire ends are for which pins so I decided to plug clock, data io, and reset, into some cheap eBay clone ST-Link.  Not connected to a computer, not connecting 3.3v and gnd, only connecting those 3 lines to the debugger caused the vPlay to wake up into the menu.  Disconnected reset from the debugger and plugged it back in, the vPlay reset again.
At least we know it'll respond to reset signals, dunno how much that effects debugging when we could just power cycle.

**Reply by Rea-XEVION** [reply]
Hey Gin, I had a thought keep popping back into my mind over the past week.
 Either last year or the year before I sold a switch game to a friend in Canada, and a shop one state over in Texas is having a clearance sale on the vPlays...

Since they're on clearance for like $9 each, I was gonna get 1x of as many flavors I can afford to hopefully find one with a different firmware than the others. Currently my cart is sitting around $130 because 12 or so flavors lol

If you'd be comfortable renting a Post Office box to recieve a package next month, I'd be more than willing to send over a handful of "front asembly only" vPlays.  Like the phone jack board without the jack, so there shouldn't be postal issues about batteries or customs issues regarding vape juice.

**Reply by ginbot86** [reply]
That would be pretty neat. It should be possible to just send it without need for a PO Box especially if only the electronics and buttons are sent...

**Reply by Rea-XEVION** [reply]
> That would be pretty neat. It should be possible to just send it without need for a PO Box especially if only the electronics and buttons are sent...

15x flavors of vPlay arrived today, already ripped the juicebox + coil wires from one that tasted like chewing on plastic, I'll pull its battery in a few days...
 Haven't yet opened every single one to check the games and animations / menu colors, but there's gonna be at least 5 or 6 for you to start poking at in a week or two.  Y'know, a couple extras for lcd breakout board or whatever other stuff

**Comment by DeepLearn6984** [comment]
Just dropped a simple hack for the VPlay20K vape. Two Python scripts are now available on GitHub: 
– vplay_rom_extractor.py – Dumps the 3 original ROMs from the device’s flash memory plus the hidden test program. 
– vplay_rom_injector.py – Allows you to replace them with your own ROMs.

GitHub repo: https://github.com/DeepLearn6984/vplay20k_hack

I'm also looking for help figuring out how to read other assets (like menu graphics or config data), and whether the mapper configuration is stored inside the SPI flash.

During my research, I found a variant of the VPlay that has a completely different menu and games (Contra, Tank 1990, Diamond, Road fighter, Excite bike, Lode runner). If anyone owns this version and can provide a flash dump, that would be super helpful.

Feedback and contributions welcome.

**Reply by ginbot86** [reply]
That is damn cool! I've yet to see the V-Play locally but I'll definitely keep my eye out for them.

**Comment by patrickniz68** [comment]
While VP vapes come with a few built-in games like Tetris, F22, and Pac-Man, they are not made to support extra games like Doom or Tamagotchi, as their hardware and software are very limited. You can try connecting your vape to a PC using a USB cable, but even then, it’s mostly for charging or firmware updates, not gaming. If you're looking for vapes with more advanced features, you might want to check out [9K Vapes](https://www.randmvape-shop.net/products/randm-tornado-9000), which are known for their smart functionality and better integration with apps and PCs.
